File: backend/TrustFlow.Api/Models/MileStone.cs
using System.Text.Json.Serialization;
using TrustFlow.Api.Models.Enums;

namespace TrustFlow.Api.Models
{
    public class MileStone
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public Guid ProjectId { get; set; }
        [JsonIgnore]
        public Project Project { get; set; } = null!;
        public string Title { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public decimal Amount { get; set; }

        public int SequenceNumber { get; set; }

        public DateTime Deadline { get; set; }

        public MileStoneStatus Status { get; set; } = MileStoneStatus.Pending;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;


    }
}

File: frontend/trustflow-web/src/features/milestones/types/milestone.ts
import type { ProjectStatus } from "@/features/projects/types/project";

export type MilestoneStatus =
  | "Pending"
  | "InProgress"
  | "Submitted"
  | "Approved"
  | "Rejected";

export type ProjectMilestone = {
  id: string;
  projectId: string;

  title: string;
  description: string;

  amount: number;
  sequenceNumber: number;

  deadline: string;
  status: MilestoneStatus;
};

export type CreateMilestoneRequest = {
  title: string;
  description: string;
  amount: number;
  sequenceNumber: number;
  deadline: string;
};
export type MilestoneStatusResponse = {
  id: string;
  status: MilestoneStatus;
};
export type ApproveMilestoneResponse = {
  id: string;
  projectId: string;
  status: MilestoneStatus;
  projectStatus: ProjectStatus;
};

export type UpdateMilestoneRequest = CreateMilestoneRequest;
